### DISCUSSION TITLE
[Feature] Resolve Inefficient Scanning of Large External Libraries for Multiple Users

### DISCUSSION THREAD

**Discussion by Peda1996** [immich-app/immich#6314] [Feature Request]
### The feature

When multiple users access a large external library through Immich, the library is rescanned for each user. This leads to inefficiency, particularly for libraries with a vast number of files.

## Steps to Reproduce
1. Set up Immich with a large external library.
2. Allow multiple users to access this library.
3. Observe that the library is rescanned for each user.

## Expected Behavior
The external library should be scanned only once, regardless of the number of users accessing it.

## Actual Behavior
The library is rescanned for each user, causing delays and inefficiency.

## Proposed Solution
Implement a mechanism where each file in the external library is hashed during the initial scan. Store these hash values in the database. When the library is accessed by a user, the system should check the database for the hash value of each file. If the hash is already present, the system should use the existing file information instead of rescanning. This approach would significantly reduce the time and resources required for multiple users to access a large library.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
There is an idea to centralize the library configuration into the admin panel, that would resolve this. https://github.com/immich-app/immich/discussions/4500#discussioncomment-7892234

**Comment by bo0tzz** [comment]
Closing this in favour of #12614
